packages/utils/src/internals/sitemap.ts

```diff
@@ -73,6 +111,26 @@ export class Sitemap {
         return parser;
     }
 
+    /**
+     * Determine the location of a sitemap file for a URL and fetch it.
+     * @param url the URL to fetch the sitemap for
+     * @param proxyUrl a proxy to be used for fetching the sitemap file
+     */
+    static async find(url: string, proxyUrl?: string): Promise<Sitemap> {
+        const sitemapUrls: string[] = [];
+
+        const sitemapUrl = new URL(url);
+        sitemapUrl.search = '';
+
+        sitemapUrl.pathname = '/sitemap.xml';
+        sitemapUrls.push(sitemapUrl.toString());
+
+        sitemapUrl.pathname = '/sitemap.txt';
+        sitemapUrls.push(sitemapUrl.toString());
+
+        return Sitemap.load(sitemapUrls, proxyUrl);
+    }
```

Comment:
I would expect this to parse the `robots.txt` (and retrieve the sitemap URL from there... and maybe use this as the last fallback step).

Consider this domain where the sitemap is not named `sitemap.xml`.

Reply:
That's good feedback. What you're describing can be accomplished with `RobotsFile.find(url).parseSitemaps()`. See [this conversation](https://github.com/apify/store-website-content-crawler/pull/212) for details on why we do this. Anyway, do you think this should somehow be incorporated into `RobotsFile.find`? Or do you have some other approach in mind?

Also, the docblock of `Sitemap.find` is very misleading, I should probably change that.

Reply:
I concur with Jindra, if we want to make this be a "take this url and find sitemaps and all urls", we should integrate robotsfile, but be careful of circular imports 😓

Reply:
I'd prefer not adding robots.txt handling here. Having two methods that do very similar things sounds like bad design to me. I'd argue that `RobotsFile.find` is the more obvious choice if you want to read sitemaps from robots.txt.

I still believe that a better name for this method might solve the problem. Something like `Sitemap.tryCommonNames` :slightly_smiling_face:

Reply:
Hmm, now I see that `RobotsFile.find` might not have been the best naming choice 😄  the thing is, `robots.txt` should always be in the website root, while sitemaps can be anywhere - having a method named `find` then makes sense only to the sitemaps to me.

On the other hand, I couldn't find the "common sitemap names" in any sitemap specification, so it feels like something that shouldn't even be here - sure, common sense is to have sitemaps named `sitemap.xml` in the website root, but there's no guarantee. I would be even for removing this altogether - it seems too simple to have a helper for it.

Reply:
I'd say lets keep it, its definitely a useful shortcut to try common names if a website doesn't expose a robots.txt file 👀 but it definitely needs a better name (like that tryCommonNames suggestion) but I've got nothing against removing it either

Reply:
Well, the functionality is simple, but kinda annoying to write, so I can see the value of the helper and I'm slightly in favor of keeping it. 

Does anyone have a better suggestion than `Sitemap.tryCommonNames`? @B4nan I'd be happy to hear your opinion as well.

Reply:
Sorry, I guess my brain is still a bit damaged, I am following the discussion and keep changing my mind. 

> I would be even for removing this altogether

It's a bit sad that this wasn't discussed in the original PR, as changing the API now is breaking, removing it is not an option.

>  the thing is, robots.txt should always be in the website root

The idea behind the `RobotsFile.find` method there was "find robots file based on any URL", also keep in mind the file is optional, but I agree it's poor naming.

---

I was initially on Honza's side, as that aligns with how this whole thing was designed before, and I'd rather not create an import cycle here. With that said, I hear the opposite arguments and I agree we should have the use case of "give me sitemaps, prefer those from robots.txt if any, but fallback to well-known sitemap locations otherwise" covered. If there is a robots file that points to a sitemap, I think it should be preferred. It doesn't make much sense to me to have a helper that would only check the well-known locations and ignore explicitly listed sitemaps from the robots file, I can already see the bug reports :]

Maybe we should just use the `robots-parser` directly in the sitemap class?

Reply:
> Note: `Sitemap.find()` is to be added with this PR, so we can still remove it :) But I'm all for keeping it, if it actually >>finds<< the sitemaps 😅

Reply:
My thoughts on this are 

1. let's keep `RobotsFile` and `RobotsFile.find` the way it is, and let's keep the functionality for loading sitemaps based on robots.txt in there - adding that to the `Sitemap` class would be pretty confusing IMO
2. regarding this PR, we can either
  a. drop it
  b. rename the method to something more appropriate
  c. change what the method does - e.g. just return the URLs so that a user can do something like `Sitemap.load([...robotsFile.getSitemaps(), ...Sitemap.guessSitemapURLs(url)])`